### PR TITLE
feat: main co-routine listens to events

### PR DESCRIPTION
**Pull request by rdash99**

### CODE REVIEW

**Comment by rdash99** [comment]
```2023-11-16 10:49:23 [2023-11-16 10:49:23,907 selector_events.py:54 __init__] Using selector: EpollSelector
2023-11-16 10:49:23 [2023-11-16 10:49:23,908 canary.py:61 watch_events] starting watcher
2023-11-16 10:49:23 [2023-11-16 10:49:23,908 canary.py:62 watch_events] ()
2023-11-16 10:49:23 [2023-11-16 10:49:23,908 canary.py:63 watch_events] {}
2023-11-16 10:49:23 [2023-11-16 10:49:23,908 canary.py:65 watch_events] listening for events
2023-11-16 10:49:23 [2023-11-16 10:49:23,929 rest.py:187 request] response body: b'{"apiVersion":"canary.ukserp.ac.uk/v1","items":[{"apiVersion":"canary.ukserp.ac.uk/v1","kind":"CanaryHTTPMonitor","metadata":{"annotations":{"meta.helm.sh/release-name":"canary","meta.helm.sh/release-namespace":"canary"},"creationTimestamp":"2023-11-14T12:48:27Z","generation":1,"labels":{"app.kubernetes.io/managed-by":"Helm"},"managedFields":[{"apiVersion":"canary.ukserp.ac.uk/v1","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:annotations":{".":{},"f:meta.helm.sh/release-name":{},"f:meta.helm.sh/release-namespace":{}},"f:labels":{".":{},"f:app.kubernetes.io/managed-by":{}}},"f:spec":{".":{},"f:interval":{},"f:status":{},"f:url":{}}},"manager":"helm","operation":"Update","time":"2023-11-14T12:48:27Z"}],"name":"github","namespace":"canary","resourceVersion":"584516","uid":"8b83deb9-7393-46ae-9e72-58b4913a1109"},"spec":{"interval":30,"status":200,"url":"https://api.github.com/octocat"}}],"kind":"CanaryHTTPMonitorList","metadata":{"continue":"","resourceVersion":"723348"}}\n'
2023-11-16 10:49:23 [2023-11-16 10:49:23,931 canary.py:23 monitor_url] starting [name='github']
2023-11-16 10:49:23 [2023-11-16 10:49:23,931 canary.py:26 monitor_url] polling [name='github'] [interval=30] [url='https://api.github.com/octocat']
2023-11-16 10:49:24 [2023-11-16 10:49:24,196 canary.py:39 monitor_url] polled [name='github'] [interval=30] [url='https://api.github.com/octocat'] [status=200] [healthy=True]```

File: src/canary.py [hunk 1]
```diff
@@ -66,9 +66,7 @@ async def watch_events(*args, **kwargs):
     tasks = dict()
 
     try:
-        # Created monitors for current objects
-        # TODO Query kubes for existing CanaryHTTPMonitor objects that are visible and iterate over them
-        # use the context manager to close http sessions automatically
+        # Create monitors for current objects
         async with ApiClient() as api:
             crds = client.CustomObjectsApi(api)
             rawmonitors = await crds.list_cluster_custom_object(group="canary.ukserp.ac.uk", version="v1", plural="canaryhttpmonitors")
```
Comment (JossWhittle) [inline src/canary.py:76]: So to confirm this line is working? And is returning the objects?

File: src/canary.py [hunk 2]
```diff
@@ -83,33 +81,46 @@ async def watch_events(*args, **kwargs):
...
-        #             monitor_url(name, url, interval, statuses)
-        #         )
+        watch = kubernetes_asyncio.watch.Watch()
+
+        # TODO Get namespace from pod
+        namespace = "canary"
+
+        while True:
+            logging.info("listening for streamed events")
+            async with watch.stream(crds.list_namespaced_custom_object, "canary.ukserp.ac.uk", "v1", namespace, "canaryhttpmonitors") as stream:
```
Comment (JossWhittle) [inline src/canary.py]: This is listing based on namespaced objects. The above query was listing cluster objects.

The CRD is currently namespaced. Can you confirm by looking in lens what the CRD you have deployed is?

File: src/canary.py (hunk 1 above)
Comment (rdash99) [reply src/canary.py:76]: Yeah, that works fine

File: src/canary.py (hunk 2 above)
Comment (rdash99) [reply src/canary.py]: Namespaced to ```canary``

**Pull request merged**